src/terminal/Grid.h

```diff
@@ -147,6 +147,59 @@ struct LogicalLine
             output += line.get().toUtf8();
         return output;
     }
+
+    // Searches from left to right, taking into account line wrapping
+    [[nodiscard]] std::optional<terminal::CellLocation> search(std::u32string_view searchText,
+                                                               ColumnOffset startPosition) const
+    {
+        // TODO Handle searchText larger than the line length
+        auto i = top;
+        for (auto line = lines.begin(); line != lines.end(); ++line)
+        {
+            auto result = line->get().search(searchText, startPosition);
+            if (result.column.has_value() && result.remainingText == 0)
+                return CellLocation { i, result.column.value() };
+            if (result.remainingText != 0)
+            {
+                auto copy = searchText;
+                copy.remove_prefix(copy.size() - result.remainingText);
+                if (line + 1 != lines.end() && (line + 1)->get().matchTextAt(copy, ColumnOffset(0)))
+                    return CellLocation { i,
+                                          ColumnOffset::cast_from(
+                                              line->get().size().value
+                                              - (searchText.size() - result.remainingText)) };
+            }
+            startPosition = ColumnOffset(0);
+            ++i;
+        }
+        return std::nullopt;
+    }
+
+    // Searches from right to left, taking into account line wrapping
+    [[nodiscard]] std::optional<terminal::CellLocation> searchReverse(std::u32string_view searchText,
+                                                                      ColumnOffset startPosition) const
+    {
+        // TODO Handle searchText larger than the line length
```

Comment:
FYI: We've a tiny CI check such that if you have comments like "// TODO(pr): ..." the CI check will fail. I personally use that to mark things as TODO and prevent accidentally merging because I forgot about it.

But i think that one here is for future reference, right? We could create'n'ref a ticket for that right away.

Reply:
Nah I'm thinking of also finishing it with this pr now. So I'll remove the comment later.